beacon-chain/blockchain/receive_block.go

```diff
@@ -237,21 +238,27 @@ func (s *Service) handleDA(
 	avs das.AvailabilityStore,
 ) (time.Duration, error) {
 	daStartTime := time.Now()
-	if avs != nil {
-		rob, err := blocks.NewROBlockWithRoot(block, blockRoot)
-		if err != nil {
-			return 0, err
-		}
 
-		nodeID := s.cfg.P2P.NodeID()
-		if err := avs.IsDataAvailable(ctx, nodeID, s.CurrentSlot(), rob); err != nil {
-			return 0, errors.Wrap(err, "could not validate blob data availability (AvailabilityStore.IsDataAvailable)")
-		}
-	} else {
+	if avs == nil {
 		if err := s.isDataAvailable(ctx, blockRoot, block); err != nil {
 			return 0, errors.Wrap(err, "is data available")
 		}
+
+		daWaitedTime := time.Since(daStartTime)
+		dataAvailWaitedTime.Observe(float64(daWaitedTime.Milliseconds()))
+		return daWaitedTime, nil
+	}
+
+	rob, err := blocks.NewROBlockWithRoot(block, blockRoot)
+	if err != nil {
+		return 0, err
 	}
+
+	nodeID := s.cfg.P2P.NodeID()
+	if err := avs.IsDataAvailable(ctx, nodeID, s.CurrentSlot(), rob); err != nil {
```

Comment:
Could you pass `nodeID` to the constructor of the `NewLazilyPersistentStoreColumn` so that it isn't necessary to change the signature of `IsDataAvailable`?

Reply:
Fixed in https://github.com/prysmaticlabs/prysm/pull/15061/commits/5c76dc6c3a96c8f7a105af8a02a6ba84a5f7e4ee.